src/Services/Catalog/Catalog.API/Extensions/RedisCacheExtensions.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System.IO;
+
+using System.Threading.Tasks;
+using System.Runtime.Serialization.Formatters.Binary;
+
+using Microsoft.Extensions.Caching.Distributed;
+using System.Collections.Generic;
+
+namespace Catalog.API.Extensions
+{
+    public static class RedisCacheExtensions
+    {
+        public static readonly string KEYS = "KEYS";
+
+        public static async Task<T> TryGetAsync<T>(this IDistributedCache cache, string key)
+        {
+            var value = await cache.GetAsync(key);
+
+            if (value != null)
+            {
+                return FromByteArray<T>(value);
+            }
+            return default(T);
+        }
+
+        public static async Task TrySetAsync<T>(this IDistributedCache cache, string key, T value)
+        {
+            if (value == null)
+                return;
+
+            T temp = await cache.TryGetAsync<T>(key);
```

Comment:
Provide a valid variable name even if its a local instantiation.

Reply:
Line 21 can be broken down as follows:
var CacheValue = FromByteArray<T>(value);
return CacheValue;